tests/pkg/gold-tester.toit

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import expect show *
+import fs
+import host.directory
+import host.file
+import host.pipe
+import host.os
+import system
+
+with-tmp-dir [block]:
+  tmp-dir := directory.mkdtemp "/tmp/test-"
+  try:
+    block.call tmp-dir
+  finally:
+    directory.rmdir --recursive tmp-dir
+
+class RunResult_:
+  stdout/string
+  stderr/string
+  exit-value/int
+
+  constructor --.stdout --.stderr --.exit-value:
+
+  exit-code -> int:
+    if (exit-value & pipe.PROCESS-SIGNALLED_) != 0:
+      // Process crashed.
+      exit-signal := pipe.exit-signal exit-value
+      return -exit-signal
+    return pipe.exit-code exit-value
+
+  normalize -> string:
+    result := stdout.replace --all "\r" ""
+    if result != "" and not result.ends-with "\n":
+      result += "\n <Missing newline at end of output> \n"
```

Comment:
Maybe be explicit about stdout/stderr here too?

```suggestion
      result += "\n <Missing newline at end of stdout> \n"
```

Reply:
done.